python/spacewalk/common/repo.py

```diff
@@ -84,7 +84,13 @@ def get(self, key: typing.Any) -> typing.Optional[typing.Any]:  # type: ignore
                 )
             return self[key]
 
-    def __init__(self, url: str, proxies: dict = None, gpg_verify: bool = True):
+    def __init__(
+        self,
+        url: str,
+        proxies: dict = None,
+        gpg_verify: bool = True,
+        timeout: typing.Optional[int] = None,
```

Comment:
Is `int` the correct type? In the python-requests [documentation](https://docs.python-requests.org/en/latest/api/#requests.request), the `timeout` parameter is documented as follows:

> timeout ([float](https://docs.python.org/3/library/functions.html#float) or [tuple](https://docs.python.org/3/library/stdtypes.html#tuple)) – (optional) How many seconds to wait for the server to send data before giving up, as a float, or a [(connect timeout, read timeout)](https://docs.python-requests.org/en/latest/user/advanced/#timeouts) tuple.

Reply:
In the context of reposync, the timeout is considered as integer value, so I guess that is what motivated the use of integer also for the timeout parameter of `DpkgRepo`  and `DebRepo` now.

There shouldn't be a problem passing this integer value to `requests` module, it should be evaluated properly.